src/main/webapp/3Dscript.js

```diff
@@ -45,144 +49,285 @@ function initThreeJs() {
+ * is a geometry instance, initializing the object with one geometry
+ * call rather than the number of data points. (O(1) vs. O(N))
+ */
+function createInstances() {
+  // Blue corresponds to the Z axis.
+  let material = new THREE.MeshBasicMaterial({color: 'blue'});
+  const geometry = new THREE.CylinderBufferGeometry(.005, .005, .1);
+  orientation.Z = new THREE.InstancedMesh(geometry, material, pose.length);
+  orientation.Z.instanceMatrix.setUsage(THREE.DynamicDrawUsage);
+  scene.add(orientation.Z);
+
+  // Green corresponds to the Z axis.
+  material = new THREE.MeshBasicMaterial({color: 'green'});
+  orientation.Y = new THREE.InstancedMesh(geometry, material, pose.length);
+  orientation.Y.instanceMatrix.setUsage(THREE.DynamicDrawUsage);
+  scene.add(orientation.Y);
+
+  // Red corresponds to the Z axis.
+  material = new THREE.MeshBasicMaterial({color: 'red'});
+  orientation.X = new THREE.InstancedMesh(geometry, material, pose.length);
+  orientation.X.instanceMatrix.setUsage(THREE.DynamicDrawUsage);
+  scene.add(orientation.X);
+}
+/**
+  * Converts latitude, longitude, and altitude coordinates (LLA) to
+  * Earth-Centered, Earth-Fixed coordinates (ECEF).
+  * @param {double} lat Latitude represented in Decimal Degrees
+  * @param {double} lng Longitude represented in Decimal Degrees
+  * @param {double} alt Altitude represented in Decimal Degrees
+  * @return {array} An array containing x, y, z ECEF coordinates.
+  */
+function llaToEcef(lat, lng, alt) {
+  // Constants.
+  const majorAxis = 6378137;
+  const flatConstant = 1 / 298.25722356;
+  const minorAxis = majorAxis*(1-flatConstant);
+  // Eccentricity.
+  const eSq = (Math.pow(majorAxis, 2) -
+    Math.pow(minorAxis, 2)) / Math.pow(majorAxis, 2);
+  const lambda = THREE.Math.degToRad(lat);
+  const phi = THREE.Math.degToRad(lng);
+  const radius = majorAxis /
+    Math.sqrt(1 - eSq * Math.pow(Math.sin(lambda), 2));
+  // Conversion.
+  const x = (radius + alt) * Math.cos(lambda) * Math.cos(phi);
+  const y = (radius + alt) * Math.cos(lambda) * Math.sin(phi);
+  const z = ((Math.pow(minorAxis, 2) * radius) /
+    Math.pow(majorAxis, 2) + alt) * Math.sin(lambda);
+
+  return [x, y, z];
+}
+
+/**
+ * Converts Earth-Centered, Earth-Fixed coordinates (ECEF) to
+ * East-North-Up coordinates (ENU).
+ * @param {double} x Latitude represented in ECEF coordinates.
+ * @param {double} y Longitude represented in ECEF coordinate.
+ * @param {double} z Altitude represented in ECEF coordinates.
+ * @param {double} lat0 The reference latitude represented in
+ *     ECEF coordinates.
+ * @param {double} lon0 The reference longitude represented in
+ *     ECEF coordinates.
+ * @param {double} h0 The reference height represented in
+ *     ECEF coordinates.
+ * @return {array} An array containing x, y, z ENU coordinates.
+ */
+function ecefToEnu(x, y, z, lat0, lon0, h0) {
+  // Converts LLA reference point to ECEF coordinates.
+  const [x0, y0, z0] = llaToEcef(lat0, lon0, h0);
```

Comment:
I think you'll end up doing a lot of repeated conversions of your origin point from LLA to ECEF. You could convert it once and pass it to this function already in ECEF.